ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -411,10 +416,25 @@ public void setNumberOfLines(int numberOfLines) {
 
   @ReactProp(name = ViewProps.LINE_HEIGHT, defaultInt = UNSET)
   public void setLineHeight(int lineHeight) {
-    mLineHeight = lineHeight == UNSET ? Float.NaN : PixelUtil.toPixelFromSP(lineHeight);
+    mLineHeightInput = lineHeight;
+    if (mAllowFontScaling) {
+      mLineHeight = lineHeight == UNSET ? Float.NaN : PixelUtil.toPixelFromSP(lineHeight);
+    } else {
+      mLineHeight = lineHeight == UNSET ? Float.NaN : PixelUtil.toPixelFromDIP(lineHeight);
+    }
+
     markUpdated();
   }
 
+  @ReactProp(name = ViewProps.ALLOW_FONT_SCALING, defaultBoolean = true)
+  public void setAllowFontScaling(boolean allowFontScaling) {
+    if (allowFontScaling != mAllowFontScaling) {
+      mAllowFontScaling = allowFontScaling;
+      setFontSize(mFontSizeInput);
+      setLineHeight(mLineHeightInput);
+    }
```

Comment:
For clarity, maybe we should call `markUpdated()` at the end of `setAllowFontScaling`. I know `setFontSize` and `setLineHeight` already call it but it'll make it clearer that this method does indeed dirty the node. Calling [`markUpdated()`](https://github.com/facebook/react-native/blob/346858a5528a95e0fe869f4fd0269aa43b31cae0/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java#L124-L133) multiple times should be cheap.